docs/tutorial/tutorial.md

```diff
@@ -0,0 +1,509 @@
+renderSquare(i) {
+  return <Square value={this.state.squares[i]} />;
+}
+```
+
+And change Square to use `this.props.value` again. Now we need to change what happens when a square is clicked. The Board component now stores which squares are filled, which means we need some way for Square to update the state of Board. Since component state is considered private, we can't update Board's state directly from Square. The usual pattern here is pass down a function from Board to Square that gets called when the square is clicked. Change `renderSquare` again so that it reads:
+
+```javascript
+return <Square value={this.state.squares[i]} onClick={() => this.handleClick(i)} />;
+```
+
+Now we're passing down two props from Board to Square: `value` and `onClick`. The latter is a function that Square can call. So let's do that by changing `render` in Square to have:
+
+```javascript
+<button className="square" onClick={() => this.props.onClick()}>
+```
+
+This means that when the square is clicked, it calls the onClick function that was passed by the parent. The `onClick` doesn't have any special meaning here, but it's popular to name handler props starting with `on` and their implementations with `handle`. Try clicking a square – you should get an error because we haven't defined `handleClick` yet. Add it to the Board class:
+
+```javascript
+handleClick(i) {
+  const squares = this.state.squares.slice();
+  squares[i] = 'X';
+  this.setState({squares: squares});
+}
+```
+
+We call `.slice()` to copy the `squares` array instead of mutating the existing array. Jump ahead a [section](/react/tutorial/tutorial.html#why-immutability-is-important) to learn why immutability is important. 
+
+Now you should be able to click in squares to fill them again, but the state is stored in the Board component instead of in each Square, which lets us continue building the game. Note how whenever Board's state changes, the Square components rerender automatically.
+
+Square no longer keeps its own state; it receives its value from its parent `Board` and informs its parent when it's clicked. We call components like this **controlled components**.
+
+## Why Immutability Is Important
+
+In the previous code example, I suggest using the `.slice()` operator to copy the `squares` array prior to making changges and to prevent mutating the existing array. Let's talk about what this means and why it an important concept to learn.
+
+There are generally two ways for changing data. The first, and most common method in past, has been to *mutate* the data by directly changing the values of a variable. The second method is to replace the data with a new copy of the object that also includes desired changes.
+
+#### Data change with mutation
+```javascript
+var player = {score:  1}
+player.score = 2 // same object mutated {score: 2}
+```
+
+#### Data change without mutation
+```javascript
+var player = {score: 1}
+player = {...player, score: 2} // new object not mutated {score: 2}
```

Comment:
This is not part of ES6. We should make it clear this syntax is a stage 3 proposal, and show an ES6 alternative.
It might also be confusing why we spread over an object with just one property which we are updating.